plugins/aws/sqs/sqsPublicAccess.js

```diff
@@ -0,0 +1,107 @@
+    more_info: 'SQS queues should be not be publically accessible to prevent unauthorized actions.',
+    recommended_action: 'Update the SQS queue policy to prevent public access.',
+    link: 'http://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-creating-custom-policies.html',
+    apis: ['SQS:listQueues', 'SQS:getQueueAttributes', 'STS:getCallerIdentity'],
+
+    run: function(cache, settings, callback) {
+        var results = [];
+        var source = {};
+        var regions = helpers.regions(settings);
+
+        var acctRegion = helpers.defaultRegion(settings);
+        var accountId = helpers.addSource(cache, source,
+            ['sts', 'getCallerIdentity', acctRegion, 'data']);
+        var awsOrGov = helpers.defaultPartition(settings);
+
+        async.each(regions.sqs, function(region, rcb){
+            var listQueues = helpers.addSource(cache, source,
+                ['sqs', 'listQueues', region]);
+
+            if (!listQueues) return rcb();
+
+            if (listQueues.err) {
+                helpers.addResult(results, 3,
+                    `Unable to query for SQS queues: ${helpers.addError(listQueues)}`, region);
+                return rcb();
+            }
+
+            if (!listQueues.data || !listQueues.data.length) {
+                helpers.addResult(results, 0, 'No SQS queues found', region);
+                return rcb();
+            }
+
+            async.each(listQueues.data, function(queue, cb){
+                var queueName = queue.substr(queue.lastIndexOf('/') + 1);
+                var resource = `arn:${awsOrGov}:sqs:${region}:${accountId}:${queueName}`;
+
+                var getQueueAttributes = helpers.addSource(cache, source,
+                    ['sqs', 'getQueueAttributes', region, queue]);
+
+                if (!getQueueAttributes ||
+                    getQueueAttributes.err ||
+                    !getQueueAttributes.data ||
+                    !getQueueAttributes.data.Attributes) {
+                    helpers.addResult(results, 3,
+                        `Unable to query attributes for queue "${queueName}"`,
+                        region, resource);
+
+                    return cb();
+                }
+
+                if (!getQueueAttributes.data.Attributes.Policy) {
+                    helpers.addResult(results, 0,
+                        `SQS queue "${queueName}" does not use a policy`,
+                        region, resource);
+                    return cb();
+                }
+
+                try {
+                    var policy = JSON.parse(getQueueAttributes.data.Attributes.Policy);
+                } catch (e) {
+                    helpers.addResult(results, 3,
+                        'The SQS queue policy is not valid JSON.',
+                        region, resource);
+
+                    return cb();
+                }
+
+                var statements = helpers.normalizePolicyDocument(policy);
```

Comment:
lets sync on this, there are a few places where we can just use the helpers.normalizePolicyDocument instead of having the try catch

Reply:
Yeah I tried this using only helpers.normalizePolicyDocument and it works. So I am changing it for you to review and if you still think that it needs discussion, we will discuss it in meeting